core/eolearn/core/utils/parsing.py

```diff
@@ -267,19 +281,10 @@ def get_features(self, eopatch: Optional[EOPatch] = None) -> List[Tuple[FeatureT
 
         If `eopatch` is not provided the method fails if an all-feature request is in the specification.
         """
-        feature_names = []
-        for feature_type, name, _ in self._feature_specs:
-            self._validate_parsing_request(feature_type, name, eopatch)
-            if name is None and feature_type.has_dict():
-                feature_names.extend(list(zip(repeat(feature_type), eopatch[feature_type])))  # type: ignore
-            else:
-                feature_names.append((feature_type, name))
-        return feature_names
+        renamed_features = self.get_renamed_features(eopatch)
+        return [feature[:2] for feature in renamed_features]  # pattern unpacking messes with typechecking
```

Comment:
Hmm, it's weird that this is necessary, how does it understand the `renamed_features` if this unpacking is not done?

Reply:
If i do `[(ftype, name) for ftype, name, _ in renamed_features]` it has `ftype: FeatureType` and `name: Optional[str]`, and it loses the link that only `FeatureType.BBOX` (and timestamp) gets a `None` value.

Reply:
Ah, sorry, my bad, I thought `[:2]` took all three elements. I'm more used to seeing `[:-1]`. Ignore this thread.